### PR TITLE
Remove basic monitor

### LINKED ISSUE #689: Remove basic monitor
I'd like to propose the removal of the `basic` monitor component, which is not used by default since a few releases ago the `pubsub` monitor was introduced.

Removing affects tests (which have setup code to run with either of them), `ipfs-cluster-service` (which can select it) and a travis stage.

Thoughts @lanzafame ?

### PR DESCRIPTION
**Pull request by kishansagathiya**
This PR removes `basic` monitor component, because it is not being
used by default since few releases ago pubsub monitor was introduced.

Fixes #689

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -28,9 +28,6 @@ jobs:
         - go test -v -coverprofile=coverage.txt -covermode=atomic ./...
       after_success:
         - bash <(curl -s https://codecov.io/bash)
-    - name: "Main Tests with basic monitor"
-      script:
-        - go test -v . -monitor basic
```
Comment (kishansagathiya) [inline .travis.yml]:
```
    - name: "Main Tests with pubsub monitor"
      script:
        - go test -v . -monitor pubsub
```

File: .travis.yml [hunk 2]
```diff
@@ -28,9 +28,9 @@ jobs:
         - go test -v -coverprofile=coverage.txt -covermode=atomic ./...
       after_success:
         - bash <(curl -s https://codecov.io/bash)
-    - name: "Main Tests with basic monitor"
+    - name: "Main Tests with pubsub monitor"
```
Comment (hsanjuan) [inline .travis.yml]: This is not necessary, the above tests are already running with pubsub

File: cluster_test.go [hunk 3]
```diff
@@ -154,9 +154,8 @@ func testingCluster(t *testing.T) (*Cluster, *mockAPI, *mockConnector, state.Sta
 
 	raftcon, _ := raft.NewConsensus(host, consensusCfg, st, false)
 
-	bmonCfg.CheckInterval = 2 * time.Second
 	psmonCfg.CheckInterval = 2 * time.Second
-	mon := makeMonitor(t, host, bmonCfg, psmonCfg)
+	mon := makeMonitor(t, host, psmonCfg)
```
Comment (hsanjuan) [inline cluster_test.go]: `makeMonitor` function is not necesary anymore, instantiate the monitor directly

File: cmd/ipfs-cluster-service/daemon.go [hunk 4]
```diff
@@ -145,7 +144,7 @@ func createCluster(
 	checkErr("creating consensus component", err)
 
 	tracker := setupPinTracker(c.String("pintracker"), host, cfgs.maptrackerCfg, cfgs.statelessTrackerCfg, cfgs.clusterCfg.Peername)
-	mon := setupMonitor(c.String("monitor"), host, cfgs.monCfg, cfgs.pubsubmonCfg)
+	mon := setupMonitor(c.String("monitor"), host, cfgs.pubsubmonCfg)
```
Comment (hsanjuan) [inline cmd/ipfs-cluster-service/daemon.go]: `setupMonitor` is not necessary anymore, instantiate the monitor directly

File: cmd/ipfs-cluster-service/main.go [hunk 5]
```diff
@@ -279,7 +279,7 @@ configuration.
 					Name:   "monitor",
 					Value:  defaultMonitor,
 					Hidden: true,
-					Usage:  "peer monitor to use [basic,pubsub].",
+					Usage:  "peer monitor to use [pubsub].",
```
Comment (hsanjuan) [inline cmd/ipfs-cluster-service/main.go]: remove this option altogether

Comment (kishansagathiya) [reply cmd/ipfs-cluster-service/main.go]: Seems like this was not being used. I didn't find `c.String("monitor")`

Comment (hsanjuan) [reply cmd/ipfs-cluster-service/main.go]: because you removed `setupMonitor` in this PR

**Pull request merged**